specs/Standards/FungibleToken/Minting.md

```diff
@@ -0,0 +1,44 @@
+# Fungible Token Minting
+
+Version `1.0.0`
+
+## Summary
+[summary]: #summary
+
+An interface for a fungible token minting, which is usually a part of a token life cycle,
+
+## Motivation
+
+Token minting can be triggered by other smart contracts. Hence we need a standard interface for minting.
+
+In a minting life cycle, token will be firstly minted and then transferred. Following scenarios are possible:
+* user can mint tokens for himself (faucet)
+* issuer will firstly mint tokens and then transfer
+* smart contract will mint tokens for a user
+
+Moreover, there are tools which filter transactions to report user tokens in a dashboard (eg the NEAR Wallet).
+These tools will usually check the first transaction. In case of NEAR Wallet it check for the `ft_transfer`, `ft_transfer_call` function calls. If a user will receive a token through minting, then it won't be reported (see Minted Tokens via Contract not showing in Wallet [#468](https://github.com/near/near-contract-helper/issues/468) issue). So, we need a standard interface in order to have a tool support.
```

Comment:
@ilblackdragon I see this paragraph might be confusing.

> you can not and should not define minting or burning standard based on function names.

As I read this standard, I don't think the goal is to enforce that all minting/burning goes through this method. It rather tries to consolidate a feature for some tokens that have a natural way of minting tokens (like faucets), so they can be used uniformly from external tools.

I agree that we should not rely on the `ft_mint` method to determine if users have some balance. In this case, I do suggest this standard enforce that the contract emits Mint events logs so that they can be used by indexers (wallet / explorer).

Reply:
>  It rather tries to consolidate a feature for some tokens that have a natural way of minting tokens (like faucets), so they can be used uniformly from external tools.

If we fix the function name just for faucets, that will confuse users even more. It should be fixed for everyone, or, if we can't do that, we should NOT declare this in NEPs.

>  In this case, I do suggest this standard enforce that the contract emits Mint events logs so that they can be used by indexers (wallet / explorer).

It's already done in https://nomicon.io/Standards/Tokens/FungibleToken/Event

Reply:
Some tokens won't expose the option to mint new tokens, and some will use a different approach. I'm thinking, for example, about bridged tokens, where you can "mint" new tokens after providing proof. I don't think this NEP is suitable for those use cases. The Mint event (log) needs to be emitted, though.

> It should be fixed for everyone

For everyone interested in exposing the minting functionality. 

In this regard @robert-zaremba, I suggest expanding the `Motivation` section with a real use case (maybe the one that motivated you to write this NEP).

Reply:
Yes, for balance tracking Events should be used.

Reply:
The goal is really for composability. For example, in Cheddar we added contract to contract authorization to allow farm for minting.